### PR TITLE
[Magiclysm] New spell Nova Flare

### PR DESCRIPTION
**Pull request by Ryan7251**
#### Summary

SUMMARY: Content "Adds a new spell called Nova Flare a better Point Flare spell."

#### Purpose of change

I was asked to to make some upgraded spells that need to use items so this is the first one.
plus you can never have to many spells.

#### Testing

I used the debug mode to give me the scrolls and i cast the spell on some NPCs it seemed to work fine but feel free to inform me of any issues in the code.

Note: I'm not sure how to fix a issue i made a pull request a few days ago that has a scroll that also uses line 80 in itemgroups/ spellbooks json and was told that may give issues not sure how to fix if both were merged but I was told I could fix it after I make the PR.

### CODE REVIEW

File: data/mods/Magiclysm/Spells/kelvinist.json [hunk 1]
```diff
@@ -23,6 +23,30 @@
...
+    "range_increment": 0.5,
+    "max_range": 11,
+    "base_energy_cost": 80,
+    "spell_class": "KELVINIST",
+    "difficulty": 8,
+    "max_level": 16,
+    "base_casting_time": 300,
+    "energy_source": "MANA"
+    },
+    {
```
Comment (BrettDong) [inline data/mods/Magiclysm/Spells/kelvinist.json:49]:
```suggestion
  },
  {
```

File: data/mods/Magiclysm/requirements/spell_components.json [hunk 2]
```diff
@@ -66,5 +66,11 @@
     "type": "requirement",
     "//": "A kilo of metal fragments suitable for the Shardspray spell",
     "components": [ [ [ "scrap", 40 ], [ "copper", 500 ], [ "gold", 200 ], [ "silver", 333 ], [ "lead", 333 ] ] ]
+  },
+  {
+      "id": "spell_components_nova_flare",
+      "type": "requirement",
+      "//": "Fire starting items suitable for the Nova Flare spell",
+      "components": [ [ [ "coal_lump", 30 ], [ "lamp_oil", 5 ] ] ]
```
Comment (BrettDong) [inline data/mods/Magiclysm/requirements/spell_components.json:74]:
```suggestion
    "id": "spell_components_nova_flare",
    "type": "requirement",
    "//": "Fire starting items suitable for the Nova Flare spell",
    "components": [ [ [ "coal_lump", 30 ], [ "lamp_oil", 5 ] ] ]
```

**Review by BrettDong — COMMENTED** [review]
JSON validation errors:

**Comment by Mrzjadacz** [comment]
Did you check if `copy-from` in spells works?
If those are just stronger versions of existing spells, it would be better to extend existing ones.

Are you planning a separate PR that adds spawns for this spell?

**Comment by Ryan7251** [comment]
> 
> 
> Did you check if `copy-from` in spells works?
> If those are just stronger versions of existing spells, it would be better to extend existing ones.
> 
> Are you planning a separate PR that adds spawns for this spell?

Sorry, but I was told to make new spells also super new to code so I got no idea what copy-from would do and how to use it.

**Comment by Mrzjadacz** [comment]
> Sorry, but I was told to make new spells also super new to code so I got no idea what copy-from would do and how to use it.

https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/JSON_INFO.md#copy-from-and-abstract

**Comment by actual-nh** [comment]
The idea is to see if adding 
```
"copy-from": "point_flare",
```
Would enable you to copy everything in `point_flare` then modify it. I doubt it will work; "copy-from" seems to be implemented separately for each type of JSON entry.

**Comment by Ryan7251** [comment]
> 
> 
> The idea is to see if adding
> 
> ```
> "copy-from": "point_flare",
> ```
> 
> Would enable you to copy everything in `point_flare` then modify it. I doubt it will work; "copy-from" seems to be implemented separately for each type of JSON entry, and none of the magic*cpp files have it.

I see thanks for clearing that up.

**Review by KorGgenT — APPROVED** [review]
had to make an itemgroup adjustment but the stats look good.

**Comment by actual-nh** [comment]
Once it's in, I'll put it up on the wiki (as I did with [Summon Wolf](http://cddawiki.chezzo.com/cdda_wiki/index.php?title=Summon_Wolf)). I don't have all the existing spells there, but I'm working on it. (I may need to ask for clarifications on some parts of the spells and/or MAGIC.md; I do plan on at least one PR already for copyediting needs I've spotted in the spells.)

**Comment by Ryan7251** [comment]
> 
> 
> had to make an itemgroup adjustment but the stats look good.

Sorry, you had to do that I'm hoping I can get this modding thing with github all fixed now that all my changed mods have been pulled.

**Pull request merged**